### ISSUE TITLE
Regression: Valkyrie Profile 2 [SLUS-21452] has solid color squares instead of some images

### ISSUE THREAD

**Issue by NetSysFire** [PCSX2/pcsx2#5588]
### Describe the Bug

This is how it currently looks:

![vp2 not expected](https://user-images.githubusercontent.com/59517351/156338955-1609c33a-9451-4cfc-b1e9-c2a66e4979ec.png)

This is how it is supposed to look:

![vp2 expected](https://user-images.githubusercontent.com/59517351/156339053-83683634-628d-4748-af92-2b1a63f7d209.png)

This also happens in software rendering mode.

**Edit: When switching from software mode and back, the 4 character images on the bottom of the screen also become those squares.**

The brown square is supposed to be a much smaller image. I rate the impact as low. The game is still perfectly playable, those squares look weird but do not impact gameplay, especially because everything is labelled anyways.

### Reproduction Steps

1. Enter any battle.
2. See the square instead of the image.

Also:

1. Enter any battle.
2. Make sure you can perform a Soul Crush.
3. Get the gauge up so you can see the Soul Crush menu after attacking.
4. All character images are also those squares.

### PCSX2 Revision

v1.7.2404 (0c98b5ff76be582d76bd64992176d643f78f86d6)

### Operating System

Linux (64bit) - Specify Distro Below

### If Linux - Specify Distro

Arch Linux

### CPU

Ryzen 1600af

### GPU

rx 570 OC version

### Logs & Dumps

[emuLog.txt](https://github.com/PCSX2/pcsx2/files/8167781/emuLog.txt)
![gs_20220302085958](https://user-images.githubusercontent.com/59517351/156320790-eff362b4-5b4b-4efd-b78d-e53eaab507fa.png)
[gs_20220302085958.gs.gz](https://github.com/PCSX2/pcsx2/files/8167790/gs_20220302085958.gs.gz)

**Comment by NetSysFire** [comment]
Bisecting done. I created this issue before bisecting because it takes ages to compile and test all of this and in the meantime maybe a great mind recognizes the issue and saves me some time. Anyways:

Git bisect says: 16cfcad0bf608e3270883750871f4c90cf2ea8ac is the first bad commit. Bisect log:

```
# bad: [49b1a496b1481b77354b20edef94e98231b4f23a] GS: Fix OSD scaling on hidpi displays
git bisect bad 49b1a496b1481b77354b20edef94e98231b4f23a
# good: [11b467e45545fec121535bd78df89eee4b2acc9e] QT: Remove 32bit build targets (which don't exist) Use a separate environment varible for our QT folder to save conflicts
git bisect good 11b467e45545fec121535bd78df89eee4b2acc9e
# bad: [e3b4d13c4720cc3a3f34ea46f51dd7ba55a35e43] HostDisplay: Remove levels/layers from HostDisplayTexture
git bisect bad e3b4d13c4720cc3a3f34ea46f51dd7ba55a35e43
# good: [e87d8da1fbc5bb59d84509d018c36ab405d469d8] GS:TC: Track whether or not textures use their clamp modes
git bisect good e87d8da1fbc5bb59d84509d018c36ab405d469d8
# good: [c45a7a2ef60bb805eb3316f9b80c374bc28e42cb] GameDB: Urban Reign
git bisect good c45a7a2ef60bb805eb3316f9b80c374bc28e42cb
# bad: [16cfcad0bf608e3270883750871f4c90cf2ea8ac] GS: fully support target rescaling in TC, take 2.
git bisect bad 16cfcad0bf608e3270883750871f4c90cf2ea8ac
# good: [ef3bc9202bc48178ca4fb7870e130ff0712b2218] memcard: don't fail on bad yaml, but this isn't the underlying issue...
git bisect good ef3bc9202bc48178ca4fb7870e130ff0712b2218
# good: [51431d22ba9d89aafe8dd601f58e89d338dc5084] memcard: fix YAML emitting bugs
git bisect good 51431d22ba9d89aafe8dd601f58e89d338dc5084
# first bad commit: [16cfcad0bf608e3270883750871f4c90cf2ea8ac] GS: fully support target rescaling in TC, take 2.
```

I will update the screenshots to provide a before/after and I am also tracking down another issue, there are some "stripes" in the game. Super hard to see on a screenshot and potentially a different issue but also popped up around the same time. The bad commit (and current master) is affected, but 51431d22ba9d89aafe8dd601f58e89d338dc5084 (I think) is not.

**Comment by iMineLink** [comment]
The dump unfortunately is useless here. Can you share a memory card with a save around the place where this occurs?
It looks to me as a failed download, this would explain also why the software mode is affected (after the hardware run).

These are the textures from which the colored square is fetched (color+alpha):

![_00281_f5000_itex_02700_P_8H_00_3ff_3ff_ff_00](https://user-images.githubusercontent.com/15114320/156416948-f0eff9ca-d2db-4e61-8bd1-49a44b90cf69.png)
![_00281_f5000_itex_02700_P_8H_00_3ff_3ff_ff_00_alpha](https://user-images.githubusercontent.com/15114320/156416951-9e194afb-81e3-4cae-944b-a3e322591381.png)

rt0 (before draw):
![_00281_f5000_rt0_01760_C_32](https://user-images.githubusercontent.com/15114320/156417116-264c03c8-53de-4d18-b7ab-4a176c6bbc36.png)
![_00281_f5000_rt0_01760_C_32_alpha](https://user-images.githubusercontent.com/15114320/156417118-ae4db2ae-b9bf-4dcb-a571-6bc495d02edc.png)

rt1 (after draw):
![_00281_f5000_rt1_01760_C_32](https://user-images.githubusercontent.com/15114320/156417122-720129a6-8605-4ce8-ab72-62c45452d3f7.png)
![_00281_f5000_rt1_01760_C_32_alpha](https://user-images.githubusercontent.com/15114320/156417123-5d053da9-5b2d-43a5-b036-b41bd10226c0.png)

rz0 (before draw):
![_00281_f5000_rz0_00000_Z_24](https://user-images.githubusercontent.com/15114320/156417125-d69a18c2-344d-420f-b439-952d8dea2e8f.png)

rz1(after draw):
![_00281_f5000_rz1_00000_Z_24](https://user-images.githubusercontent.com/15114320/156417128-d4d19466-c70f-49bc-a6a3-3cc57542a36b.png)

**Comment by NetSysFire** [comment]
Thanks for the testing. I can provide one when I am at home again where my desktop is. This affects any battle, so if you start a new game you should be able to click through the intro and get to the forest where a battle will soon follow. This should not take much time.

I don't remember exactly anymore but you start in Solde where you have to go to the pub (on the right side of town) to fetch rufus, then you can leave the city (left side) and can go to the forest where a battle is in arms reach, you can skip the tutorial and you should see what I see. Just walk into the shadowy entities for a battle to start.

**Comment by NetSysFire** [comment]
Here is a memory card containing my save: [Mcd001.ps2.gz](https://github.com/PCSX2/pcsx2/files/8179298/Mcd001.ps2.gz)
I had to gzip because github does not like file extensions.

If you load the save you will be on the world map and you can just enter Lost Forest and enter a battle easily. It affects any battle, but Lost Forest is the first (and easiest) battle location of the game, so I chose that.

**Comment by NetSysFire** [comment]
I apologize, I think I found the cause. I left auto flush on after I used it to test another game. I will close this, but since I also bisected and tested with auto flush on, it appears to have changed its behavior in the mentioned commit. This may or may not be a bug, this is up for you to decide. Feel free to reopen if this is actually one.

**Comment by refractionpcsx2** [comment]
If it works now with autoflush on, I'm happy. Auto Flush *shouldn't* break anything.

**Comment by NetSysFire** [comment]
No, I meant the opposite. I left auto flush on. I disabled it and now those stripes I mentioned and the bug this issue is about is gone. I will build the latest master (takes some time) and report back how it looks with autoflush on or off.

I mean these stripes, they are really hard to see and are quite possibly the cause of auto flush.
![vp2 stripes](https://user-images.githubusercontent.com/59517351/158039467-b8e79720-b7e1-419e-b7ac-c54a90abb5c0.png)

**Comment by refractionpcsx2** [comment]
Oh, well it's likely the game requires auto flush, if somebody else could confirm, then we could turn it on.  That does look like a case where auto flush could be required.

**Comment by refractionpcsx2** [comment]
Can you provide a GS dump of that section? I can't get Auto Flush to trigger at the start of the game.

**Comment by NetSysFire** [comment]
I apologize again. Further testing revealed the root cause of the missing pictures was Preload Frame Data, not Auto Flush, I mixed it up. Auto Flush still introduces those stripes though. I can reproduce in multiple games and in almost any section of the game.

To be very clear here: **Turning off Auto Flush makes it look as expected. The stripes vanish. Auto Flush is the cause of those stripes.**

Kythena Plains - chosen solely because the visual effects in that area make those very visible.

![gs_20220313015541_Valkyrie Profile 2 - Silmeria _NTSC-U__SLUS-21452](https://user-images.githubusercontent.com/59517351/158040340-b85c7cea-400e-4364-9da0-e9ebd765f4ad.png)
[gs_20220313015541_Valkyrie Profile 2 - Silmeria _NTSC-U__SLUS-21452.gs.gz](https://github.com/PCSX2/pcsx2/files/8238785/gs_20220313015541_Valkyrie.Profile.2.-.Silmeria._NTSC-U__SLUS-21452.gs.gz)

World map - the one you requested.

![gs_20220313015508_Valkyrie Profile 2 - Silmeria _NTSC-U__SLUS-21452](https://user-images.githubusercontent.com/59517351/158040361-bcdcc959-448a-4fb4-9812-9990ba4cd9ee.png)
[gs_20220313015508_Valkyrie Profile 2 - Silmeria _NTSC-U__SLUS-21452.gs.gz](https://github.com/PCSX2/pcsx2/files/8238786/gs_20220313015508_Valkyrie.Profile.2.-.Silmeria._NTSC-U__SLUS-21452.gs.gz)

GS configs, so there are no more mix ups (github does not like inis, so I just changed the extension):
[GSdx.txt](https://github.com/PCSX2/pcsx2/files/8238790/GSdx.txt)
[GS.txt](https://github.com/PCSX2/pcsx2/files/8238791/GS.txt)

**Comment by refractionpcsx2** [comment]
What version of PCSX2 are you using?

**Comment by NetSysFire** [comment]
I just built the latest master. How do I know if it is triggering? When I turn it off and reload the save, it looks as expected.

![gs_20220313022457_Valkyrie Profile 2 - Silmeria _NTSC-U__SLUS-21452](https://user-images.githubusercontent.com/59517351/158040879-e0330cf6-832f-40e5-8298-8ebb404bf494.png)
[gs_20220313022457_Valkyrie Profile 2 - Silmeria _NTSC-U__SLUS-21452.gs.gz](https://github.com/PCSX2/pcsx2/files/8238819/gs_20220313022457_Valkyrie.Profile.2.-.Silmeria._NTSC-U__SLUS-21452.gs.gz)

Yes, those artifacts at the right side of the screen are new but do not appear when autoflush is enabled. They get a bit annoying in some areas (fairly sure they started appearing in Dipan) and vanish when using software mode.
This may or may not be a regression but honestly I do not know. I know those were not there in the past. But one thing at a time.

**Comment by refractionpcsx2** [comment]
Hmm well the only flush it's triggering is TEXFLUSH (which it should do), but I can't tell with a single frame dump, it looks identical,. Can you do a multiframe dump please?

**Comment by NetSysFire** [comment]
I just discovered, since I always skip it lol, that even the opening cutscene is affected. So I made a multiframe dump of this. The stripes are a bit harder to see but still there if you look closely.

![vp2 cutscene stripes](https://user-images.githubusercontent.com/59517351/158041719-951c811d-ad37-4942-8b00-59ff0ab6d418.png)

![gs_20220313025653_(2)_Valkyrie Profile 2 - Silmeria _NTSC-U__SLUS-21452](https://user-images.githubusercontent.com/59517351/158041638-256a278e-cdcf-4f4f-87a1-c106e86a8707.png)

It is really annoying that github does not support .xz because the gzipped variant is too big to upload to github, so I uploaded the .xz to a temporary file hosting service which is unfortunately necessary. It might be gone soon.
https://0x0.st/oZfU.xz

**Comment by refractionpcsx2** [comment]
You realise you're pointing out an FMV, right?

**Comment by NetSysFire** [comment]
Oh is this bad? I thought its good because its really early in the game (and 2D, therefore maybe isolating the issue?) and therefore good to try and reproduce. I can go with Kythena Plains again if thats better.

**Comment by refractionpcsx2** [comment]
I cannot reproduce that issue on any of your dumps, to me it looks like you turned "texture filtering" off in the shader options and it's just scaling to the screen badly or something, I don't see how this is autoflush at fault.

I even tried my EU copy of VP2 and couldn't replicate it.

**Comment by NetSysFire** [comment]
Texture filtering of display is on, I tried turning it off just now to test if the behavior changes, but it did not. This is super weird. The GS dump shows those vertical stripes, too. If I turn off autoflush, those stripes vanish. Anything else I could test? The logical conclusion for me is to leave autoflush off, but I am willing to debug why it causes those stripes since you said it should not break anything.

I can confirm these stripes appearing in other games, too (I tried Jak and Dexter II which is affected by this but Odin Sphere is not it appears) when autoflush is enabled. As soon as I turn it off, everything is normal again. I could try more games but it is 3:30 AM but I am getting physically tired. I can open another issue for this, if this is even a bug, since the original issue I had is solved because it was simply an error on layer 8.

**Comment by refractionpcsx2** [comment]
The only other thing it could be is a barrier problem on your GPU, since auto flush will cause more barriers to happen. When did you last update your GPU drivers?

**Comment by NetSysFire** [comment]
I use Linux and the GPU driver (amdgpu) is shipped with the kernel. I do not update the kernel often due to me being lazy. I think I last updated it two weeks ago? Maybe it is a regression. I could try different versions after a night of sleep, thanks for the hint. GPU regressions do happen and are not uncommon unfortunately.

**Comment by refractionpcsx2** [comment]
Could be worth a try, maybe try a mesa driver or something? i dunno, I'm afraid I can't reproduce the issue here.

**Comment by NetSysFire** [comment]
I am definitely not a GPU pro, but I think I already am using mesa. amdgpu is making sure the card works and outputs video, but mesa is the opengl and vulkan implementation. The culprit is often times just amdgpu though. I will report back after testing if an older kernel (and therefore older amdgpu version) works. If so, this is a kernel issue and not related to PCSX2 anymore. Thanks for all the hints and patience so far.

**Comment by NetSysFire** [comment]
I have downgraded as far as I could without breaking my system and did not observe any change. It may or may not be mesa (it is definitely not amdgpu though), but I did some more testing and it *appears* to affect 2D sprites only. Without some major effort on my part (building in a chroot and all that), I can not debug this further and I will simply give up on that unless me building an ancient pcsx2 version shows it is actually an emulator issue. If this is the case, I will open another issue for that.

The only other thing left to check out right now is the original issue, me having Preload Frame Data enabled. If this is known for breaking things, fair enough. If not, the bisect shows it has changed behavior which may or may not be a bug.

**Comment by NetSysFire** [comment]
@refractionpcsx2 I found the cause of the issue we tried so hard to debug. I even reset my configs but apparently it did not get all.

The specific settings that caused these artifacts were `UserHacks_HalfPixelOffset = 2` and `UserHacks_round_sprite_offset = 2` (which is visible in the configs I shared). Why autoflush made these visible, I have no idea. I might have unchecked the manual hacks checkbox because I *assumed* (there it is, another error on layer 8) autoflush was the only hack I have enabled (which did not only disable autoflush but the other two options as well). After two hours of trying old builds (and more trying old mesa and kernel versions) to see if it was a regression and actually clearing `~/.config/PCSX2`, I found that the issue is actually the config.

Resetting the configs via the UI does not appear to actually get GS.ini which may or may not be intended.
